packages/devtools_extensions/lib/src/model.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'api.dart';
+
+/// Data model for a devtools extension event that will be sent and received
+/// over 'postMessage' between DevTools and the embedded extension iFrame.
+class DevToolsExtensionEvent {
+  DevToolsExtensionEvent(this.type, {this.data});
+
+  factory DevToolsExtensionEvent.parse(Map<String, Object?> json) {
+    final eventType =
+        DevToolsExtensionEventType.from(json[_typeKey]! as String);
+    final data = (json[_dataKey] as Map?)?.cast<String, Object?>();
+    return DevToolsExtensionEvent(eventType, data: data);
+  }
+
+  static DevToolsExtensionEvent? tryParse(Object data) {
+    try {
+      final dataAsMap = (data as Map).cast<String, Object?>();
+      return DevToolsExtensionEvent.parse(dataAsMap);
+    } catch (_) {
+      return null;
+    }
+  }
+
+  static const _typeKey = 'type';
+  static const _dataKey = 'data';
+
+  static DevToolsExtensionEvent ping =
+      DevToolsExtensionEvent(DevToolsExtensionEventType.ping);
+
+  static DevToolsExtensionEvent pong =
+      DevToolsExtensionEvent(DevToolsExtensionEventType.pong);
+
+  final DevToolsExtensionEventType type;
```

Comment:
What's the point to differentiate types of extensions if they behave equally?

Reply:
Each extension will share some underlying framework for integration with DevTools (connection to DevTools, connection to a VM service, etc.), but the functionality that each extension provides will be entirely dependent on the use case a package or extension author addresses.